lib/system.ml

```diff
@@ -90,6 +90,7 @@ let process_subdirectories f path =
 let warn_cannot_open_dir =
   CWarnings.create ~name:"cannot-open-dir" ~category:CWarnings.CoreCategories.filesystem
   (fun dir -> str ("Cannot open directory " ^ dir))
+let warn_cannot_open_dir ?loc d = warn_cannot_open_dir ?loc d
```

Comment:
It looks annoying to have to do this everywhere a warning is exposed, how about instead leave the type of `CWarnings.create` as before and add a `CWarning.create_with_qf` that has the new type?

Reply:
ok, on it